model/errors/data_source_io_error.go

```diff
@@ -0,0 +1,17 @@
+package errors
+
+import (
+    "fmt"
+)
+
+type DataSourceIOError struct {
+    ErrMsg error
+}
+
+func (e DataSourceIOError) Error() string {
+    return fmt.Sprintf("Error while manipulating data source")
+}
+
+func (e DataSourceIOError) Unwrap() error {
+    return e.ErrMsg
+}
```

Comment:
This file should be moved out from `/model/errors` package, into its own `/errors` package.

Normally `models` is used for your entities (although an error could be considered as an entity) but IMHO and in practice, makes more sense to be in it's own package.